### PR TITLE
PWN-7623 Get bundle for claim

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7623

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -29,6 +29,7 @@
         android:fullBackupContent="@xml/backup_rules"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
+        android:largeHeap="true"
```
Comment (gslevinkov) [inline app/src/main/AndroidManifest.xml:32]: revert?

File: app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+import org.p2p.wallet.bridge.model.BridgeBundle
 import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
 
 class ClaimInteractor(
     private val ethereumClaimRepository: EthereumClaimRepository,
     private val ethereumRepository: EthereumRepository,
     private val tokenKeyProvider: TokenKeyProvider
-)
+) {
+    suspend fun getEthereumBundle(erc20TokenAddress: String, amount: String): BridgeBundle {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt:14]: i think erc20Token address is hex?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt:14]: amount -> amountInLamports

File: app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt [hunk 3]
```diff
@@ -36,6 +49,27 @@ class ClaimPresenter(
...
+            try {
+                val bundle = claimInteractor.getEthereumBundle(
+                    tokenToClaim.publicKey,
+                    totalToClaim.toString()
+                )
+                val timeExpires = bundle.expiresAt.toZonedDateTime().dateMilli()
+                val timerToSet = timeExpires - System.currentTimeMillis()
+                startRefreshJob(timerToSet)
+            } catch (error: Throwable) {
+                view?.showErrorMessage(error)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/claim/ui/ClaimPresenter.kt:66]: Timber.e

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt (hunk 2 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt:14]: hex is String =)

Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt:14]: @gslevinkov erc20Token will be in EthAddress wrap class

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt:14]: tokenToClaim.publicKey, 

tokenToClaim == Token.Eth

Token.Eth publicKey == String

data class Eth constructor(
        override val publicKey: String,
        val totalInUsd: BigDecimal?,
        val total: BigDecimal,
        override val tokenSymbol: String,
        override val decimals: Int,
        override val mintAddress: String,
        override val tokenName: String,
        override val iconUrl: String?,
        override var rate: BigDecimal?,
        override var currency: String = Constants.USD_READABLE_SYMBOL
    )
    
    you want to make it EthAddress ? or wrap it in presenter from string?

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**